### ISSUE TITLE
Uncaught TypeError: Cannot read property 'split' of undefined

### ISSUE THREAD

**Issue by takanorip** [Keyang/node-csvtojson#185]
I get this error when using csvtojson. 
My node version is `v7.10.0`.

```
Converter.js:85 Uncaught TypeError: Cannot read property 'split' of undefined
    at Object.<anonymous> (Converter.js:85)
    at Object.module.exports (Converter.js:542)
    at __webpack_require__ (bootstrap 6e0e07b…:659)
    at fn (bootstrap 6e0e07b…:85)
    at Object.<anonymous> (index.js:2)
    at __webpack_require__ (bootstrap 6e0e07b…:659)
    at fn (bootstrap 6e0e07b…:85)
    at Object.<anonymous> (csv2json.js:2)
    at __webpack_require__ (bootstrap 6e0e07b…:659)
    at fn (bootstrap 6e0e07b…:85)
```

When I import csvtojson, this error occur.
```
import CSV from 'csvtojson';
```

**Comment by JoseExposito** [comment]
Same. It happens because Converter.js is using `process.versions.node` here:

https://github.com/Keyang/node-csvtojson/blob/46f97473b6ab68d40f82e5217668beab51b4e183/libs/core/Converter.js#L85

Which is not available using a bundler like webpack (in my case). @Keyang, this can by fixed by checking if the function is available:

```
function bufFromString(str){
    return (typeof Buffer. from == 'function')
        ? Buffer.from(str,"utf8")
        : new Buffer(str,"utf8")
}
```

You can quickly check the behaviour:

```
$ node --version
v7.6.0
$ node
> (typeof Buffer. from == 'function')
true
```

@Kaylang, should I create a pull request?

**Comment by Keyang** [comment]
@JoseExposito Thanks for pointing out.
It was using the same way as you mentioned. However, this seems not working with node.v4 which has `Buffer.from` with a different signature. Thus I changed to using version number.

Please create a pull request and we will go from there.

Thank you.

~Keyang

**Comment by JoseExposito** [comment]
Very true, the signature is there in older versions, ignore my previous solution as it won't work. However, after checking the documentation for Node 4 & 8:

(Node 4) https://nodejs.org/docs/latest-v4.x/api/buffer.html
(Node 8) https://nodejs.org/api/buffer.html

I think this should work:

```
$ node
> let str = 'Hello World!'
undefined
> let buffer = Buffer.allocUnsafe(str.length)
undefined
> buffer.write(str)
12
> buffer.toString('utf8')
'Hello World!'
```

Note that `allocUnsafe` is secure here as we just write in the buffer after allocating it and is faster that `alloc`.

Creating pull request...

**Comment by jjperezaguinaga** [comment]
Just a ping to @Keyang that the PR from @JoseExposito does work... currently using it on my project, although I haven't checked why build ain't green

**Comment by CoIin** [comment]
Do you know what build the 'split' error will be fixed in? Still getting it in version 0.12.1.

**Comment by serenanubi** [comment]
I'm guessing the build isn't green partly because there are no eol characters (;).

**Comment by JoseExposito** [comment]
Sorry for the delay guys, I forgot about this pull request.

I just added full support for very old Node.js version (<= 5.x) so hopefully @Keyang feels more comfortable merging https://github.com/Keyang/node-csvtojson/pull/186 into master now that all the tests pass.

**Comment by wzup** [comment]
What is this?

![node inspector - google chrome 2017-10-03 13 34 38](https://user-images.githubusercontent.com/3880497/31111348-f08607e2-a83f-11e7-8c8e-ebdb2abbb4ed.png)

**Comment by JoseExposito** [comment]
It looks like the change is already available in 1.1.8, please update.
